stale-bot: Make its message more instructive

doronbehar opened the pull request:
###### Motivation for this change

Address the comments the stale bot got at https://github.com/NixOS/rfcs/pull/51 after it's deployment.

I know the stale bot is using the same message for both issues and PRs. My suggestion improve the bot's message for PRs, and it's not suitable for issues. I hope it's possible to configure a different message for PRs vs issues and I think we can't find a single message that will fit both cases. This message is also longer, but I think it explains better what's the situation and it gives better suggestions as to what can be done. Also, it gives less of an impression that it's the PR's author's fault that their PR was marked as stale. Other improvements include the bot saying that it won't close the PR, along with a fix to the link to IRC (I'm not sure if it was rendered as expected before :/).

EDIT: I removed the copy of it rendered as it will change often now along with reviews.

doronbehar committed:
stale-bot: Make it's messages more instructive

As somewhat discussed at: https://git.io/JJkRa, make the bot's message a
bit more instructive, and clearer as for what it will and won't do, so
random contributors will have a better idea what this is about.

Also: Split stale issues' and PRs' messages because each should
suggest different actions to do.

ryantm commented:
I looked a little at the stale bot code and it does not appear possible to have different messages for issues and PRs.

abathur commented:
> I hope it's possible to configure a different message for PRs vs issues

The stale bot[ _does_ appear to support separate messages](https://github.com/probot/stale#usage) (or any other setting) for PRs and issues. It'd look something like:

```yaml
pulls:
  markComment: |
    PR comment...

issues:
  markComment: |
    issue comment...
```

ryantm commented:
I think I wasn't looking closely enough! Thanks, @abathur.

abathur commented:
> This message is also longer, but I think it explains better what's the situation and it gives better suggestions as to what can be done. Also, it gives less of an impression that it's the PR's author's fault that their PR was marked as stale. Other improvements include the bot saying that it won't close the PR, along with a fix to the link to IRC

@ryantm @doronbehar I see some ways I think I can improve the style and structure of this message, but I want to avoid distracting from the more important task of deciding if this message says the right things. Just wanted to note that I'll try to be available to rephrase if either of you @ me. https://github.com/NixOS/rfcs/pull/51#issuecomment-636921555 has a rough reference of how I'd be rephrasing.

Ma27 reviewed (approved):
Please update the config to only apply this for PRs (as @abathur suggested). Apart from that I really like the change :+1:

bhipple commented on .github/stale.yml:
```diff
@@ -10,16 +10,18 @@ exemptLabels:
-  1. Search for maintainers and people that previously touched the related code and @ mention them in a comment.
-  2. Ask on the [NixOS Discourse](https://discourse.nixos.org/).
-  3. Ask on the [#nixos channel](irc://irc.freenode.net/#nixos) on [irc.freenode.net](https://freenode.net).
+  If your PR has had reviews and never the less got stale, make sure you've responded to all of the reviewer's requests / questions. Usually when PR authors show responsibility and dedication, reviewers (privileged or not), show dedication as well. If you've pushed a change, it's possible the reviewer wasn't notified about your push via email, so you can always [officially request them for a revi ...
```
```suggestion
  If your PR has had reviews and nevertheless got stale, make sure you've responded to all of the reviewer's requests / questions. Usually when PR authors show responsibility and dedication, reviewers (privileged or not), show dedication as well. If you've pushed a change, it's possible the reviewer wasn't notified about your push via email, so you can always [officially request them for a review](https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/requesting-a-pull-request-review), or just @ mention them and say you've addressed their comments.
```

bhipple reviewed (approved):
Looks great to me; have a couple minor editing / grammar points.

doronbehar commented:
Thank you all for the reviews. Since we are taking this as the new _PR_ stale message, I hope it's OK to suggest while at it a better message for issues? @abathur I took inspiration from your proposal at https://github.com/NixOS/rfcs/pull/51#issuecomment-636921555 for the issues' message. In my last push, I addressed @bhipple  and @Ma27's comments, and added this issues message:

-----

Hello, I'm a bot and I thank you in the name of the community for opening this issue.

To help our human contributors focus on the most-relevant reports, I check up on old issues to see if they're still relevant. This issue has had no activity for 180 days, and so I marked it as stale, but you can rest assured it will never be closed by a non-human.

The community will appreciate your effort, if you'd recheck the issue, and hopefully close it if you see it is no longer valid.

If the issue persists, and you'd like to remove the stale label, you simply need to leave a comment. Your comment can be as simple as "still important to me". If you'd like it to get more attention, you can ask for help by searching for maintainers and people that previously touched related code and @ mention them in a comment. You can use [Git blame](https://git-scm.com/docs/git-blame) or [GitHub's web interface](https://docs.github.com/en/github/managing-files-in-a-repository/tracking-changes-in-a-file) on the relevant files to find them.

Lastly, you can always ask for help at [our Discourse Forum](https://discourse.nixos.org/) or at [#nixos' IRC channel](https://webchat.freenode.net/#nixos).

ryantm commented on .github/stale.yml:
```diff
@@ -9,17 +9,33 @@ exemptLabels:
+
+    To help our human contributors focus on the most-relevant reports, I check up on old issues to see if they're still relevant. This issue has had no activity for 180 days, and so I marked it as stale, but you can rest assured it will never be closed by a non-human.
+
+    The community will appreciate your effort, if you'd recheck the issue, and hopefully close it if you see it is no longer valid.
```
```suggestion
    The community would appreciate if you'd check if the issue is still valid, and, if it isn't, close it.
```

doronbehar replied:
@ryantm I like the usage of the word "effort" because sometimes it can be quiet an effort to set up the environment or whatever needed to reproduce an issue. So I'm pushing:

The community would appreciate your effort, if you'd check if the issue is still valid, and if it isn't, close it.

ryantm replied:
Okay, then I think it should be:

The community would appreciate your effort in checking if the issue is still valid, and if it isn't, close it.

ryantm replied:
That's kind of clunky though, maybe better as two sentences:

"The community would appreciate your effort in checking if the issue is still valid. If it isn't, please close it.

doronbehar replied:
That's better.

ryantm commented on .github/stale.yml:
```diff
@@ -9,17 +9,33 @@ exemptLabels:
+    
+    If your PR wasn't reviewed at all, it might help to find someone who's perhaps a user of the package or module you are changing, or alternatively, ask once more for a review by the maintainer of the package/module this is about. If you don't know any, you can use [Git blame](https://git-scm.com/docs/git-blame) on the relevant files, or [GitHub's web interface](https://docs.github.com/en/githu ...
+
+    If your PR has had reviews and nevertheless got stale, make sure you've responded to all of the reviewer's requests / questions. Usually when PR authors show responsibility and dedication, reviewers (privileged or not), show dedication as well. If you've pushed a change, it's possible the reviewer wasn't notified about your push via email, so you can always [officially request them for a revi ...
```
```suggestion
    If your PR has had reviews and nevertheless got stale, make sure you've responded to all of the reviewer's requests / questions. Usually when PR authors show responsibility and dedication, reviewers (privileged or not) show dedication as well. If you've pushed a change, it's possible the reviewer wasn't notified about your push via email, so you can always [officially request them for a review](https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/requesting-a-pull-request-review), or just @ mention them and say you've addressed their comments.
```

doronbehar replied:
:+1:.
